nifi-nar-bundles/nifi-snowflake-bundle/nifi-snowflake-services/src/main/java/org/apache/nifi/snowflake/service/SnowflakeComputingConnectionPool.java

```diff
@@ -0,0 +1,265 @@
+     * made since the underlying system may still go off-line during normal
+     * operation of the connection pool.
+     *
+     * @param context the configuration context
+     * @throws InitializationException if unable to create a database connection
+     */
+    @OnEnabled
+    public void onConfigured(final ConfigurationContext context) throws InitializationException {
+        final String snowflakeUrl = context.getProperty(SNOWFLAKE_URL).evaluateAttributeExpressions().getValue();
+        final String connectionString;
+        if (snowflakeUrl.startsWith("jdbc:snowflake")) {
+            connectionString = snowflakeUrl;
+        } else {
+            connectionString = "jdbc:snowflake://" + snowflakeUrl;
+        }
+        final String user = context.getProperty(SNOWFLAKE_USER).evaluateAttributeExpressions().getValue();
+        final String password = context.getProperty(SNOWFLAKE_PASSWORD).evaluateAttributeExpressions().getValue();
+
+        final Integer maxTotal = context.getProperty(MAX_TOTAL_CONNECTIONS).evaluateAttributeExpressions().asInteger();
+        final String validationQuery = context.getProperty(VALIDATION_QUERY).evaluateAttributeExpressions().getValue();
+        final Long maxWaitMillis = extractMillisWithInfinite(context.getProperty(MAX_WAIT_TIME).evaluateAttributeExpressions());
+        final Integer minIdle = context.getProperty(MIN_IDLE).evaluateAttributeExpressions().asInteger();
+        final Integer maxIdle = context.getProperty(MAX_IDLE).evaluateAttributeExpressions().asInteger();
+        final Long maxConnLifetimeMillis = extractMillisWithInfinite(context.getProperty(MAX_CONN_LIFETIME).evaluateAttributeExpressions());
+        final Long timeBetweenEvictionRunsMillis = extractMillisWithInfinite(context.getProperty(EVICTION_RUN_PERIOD).evaluateAttributeExpressions());
+        final Long minEvictableIdleTimeMillis = extractMillisWithInfinite(context.getProperty(MIN_EVICTABLE_IDLE_TIME).evaluateAttributeExpressions());
+        final Long softMinEvictableIdleTimeMillis = extractMillisWithInfinite(context.getProperty(SOFT_MIN_EVICTABLE_IDLE_TIME).evaluateAttributeExpressions());
+
+        dataSource = new BasicDataSource();
+
+        dataSource.setDriver(getDriver(SnowflakeDriver.class.getName(), connectionString));
+
+        dataSource.setUrl(connectionString);
+        dataSource.setUsername(user);
+        dataSource.setPassword(password);
+
+        dataSource.setMaxWaitMillis(maxWaitMillis);
+        dataSource.setMaxTotal(maxTotal);
+        dataSource.setMinIdle(minIdle);
+        dataSource.setMaxIdle(maxIdle);
+        dataSource.setMaxConnLifetimeMillis(maxConnLifetimeMillis);
+        dataSource.setTimeBetweenEvictionRunsMillis(timeBetweenEvictionRunsMillis);
+        dataSource.setMinEvictableIdleTimeMillis(minEvictableIdleTimeMillis);
+        dataSource.setSoftMinEvictableIdleTimeMillis(softMinEvictableIdleTimeMillis);
```

Comment:
As stated before if I keep the new propertyDescriptors (to be able to have separate name and displayName for them) then the shared code would require a refactor that would introduce the same amount of complexity we try to prevent with the refactor in the first place.
I could create a method something like setupDatasource but it would get a bunch of parameters or would get a DTO which I would need to call getters and setters on - just like on the datasource itself.

Again, it's not business logic but a sequence of getter and setter calls. No need to overengineer it.

I'm willing to reconsider if I saw a specific suggestion and the rest of the community (at least one more contributor) would vote for that implementation.